doc/cookbook/type/create_field_type_extension.rst

```diff
@@ -0,0 +1,220 @@
+.. index::
+   single: Type; Field type extension
+
+How to Create a Field Type Extension
+====================================
+
+:doc:`Custom search field types <create_custom_field_type>` are great when
```

Comment:
i am a bit overwhelmed, will take a break. Please refer to this line link to continue perhaps later or tomorrow :older_man: